internal/clusterconfig/recert.go

```diff
@@ -0,0 +1,92 @@
+package clusterconfig
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/clusterinfo"
+	"github.com/openshift-kni/lifecycle-agent/utils"
+)
+
+const recertConfigFile = "recert_config.json"
+
+type recertConfig struct {
+	DryRun            bool     `json:"dry_run,omitempty"`
+	ExtendExpiration  bool     `json:"extend_expiration,omitempty"`
+	EtcdEndpoint      string   `json:"etcd_endpoint,omitempty"`
+	ClusterRename     string   `json:"cluster_rename,omitempty"`
+	SummaryFile       string   `json:"summary_file,omitempty"`
+	StaticDirs        []string `json:"static_dirs,omitempty"`
+	StaticFiles       []string `json:"static_files,omitempty"`
+	CNSanReplaceRules []string `json:"cn_san_replace_rules,omitempty"`
+	UseKey            []string `json:"use_key,omitempty"`
+	UseCert           string   `json:"use_cert,omitempty"`
+}
+
+// CreateRecertConfigFile function to create recert config file
+// those params will be provided to an installation script after reboot
+// that will run recert command with them
+func CreateRecertConfigFile(clusterData *clusterinfo.ClusterInfo,
+	clusterConfigPath string, ostreeDir string) error {
+	// read seed manifest json
+	seedManifestFile := filepath.Join(filepath.Dir(clusterConfigPath), seedManifest)
+	seedClusterInfo := &clusterinfo.ClusterInfo{}
+	err := utils.ReadYamlOrJSONFile(seedManifestFile, seedClusterInfo)
+	if err != nil {
+		return fmt.Errorf("failed to parse %s data, error: %w", seedManifestFile, err)
+	}
+
+	config := recertConfig{
+		DryRun:           false,
+		EtcdEndpoint:     "localhost:2379",
+		ExtendExpiration: true,
+		StaticDirs:       []string{"/kubelet", "/kubernetes", "/machine-config-daemon"},
+		StaticFiles:      []string{"/host-etc/mcs-machine-config-content.json"},
+		ClusterRename:    fmt.Sprintf("%s:%s", clusterData.ClusterName, clusterData.Domain),
+		SummaryFile:      "/var/tmp/recert-summary.yaml",
```

Comment:
Is `/var/tmp` mounted?

Reply:
We also need a clean summary file, so we can include it in must-gather

Reply:
Yep, it is mounted

Reply:
Alright, what about:

>We also need a clean summary file, so we can include it in must-gather

Reply:
https://docs.google.com/presentation/d/10_SAZpbc2Ri1w__89K6EuithAK0ta6qG_wV-4tjqwXo/edit#slide=id.g294d0ae4bf8_0_137

Reply:
I see, not sure where to add it currently, let me open ticket for it